### DISCUSSION TITLE
React Native-Web3Modal Metamask Connection Problem

### DISCUSSION THREAD

**Discussion by Lestonz** [WalletConnect/walletconnect-monorepo#2916] [Developer Support] [answered]
Can someone help me about web3modal?
1-Metamask is not sending a connection request to the wallet.
2-Turst W. sends connection request but gives another error.
3. Rainbow sends the connection request and confirms it's super but "close func" doesn't work.

I share some error videos from Twitter link: https://twitter.com/lestonz/status/1674866499004030988?s=20

**Comment by ignaciosantise** [comment, marked as answer]
hey @Lestonz 👋  Those issues are on the wallet side. As i can see from your video, you are using this sessionParams:

```
export const sessionParams = {
  namespaces: {
    eip155: {
      methods: [
        'eth_sendTransaction',
        'eth_sign',
        'personal_sign',
        'eth_signTypedData',
      ],
      chains: ['eip155:1', 'eip155:5'],
      events: ['chainChanged', 'accountsChanged'],
      rpcMap: {},
    },
  },
};
```

1. MetaMask: We are currently working with them to fix their multichain issue. 
2. Trust: They don't support Goerli Network.
3. Rainbow Wallet connected successfully, what's the error there?

We are also reaching wallets to support `optionalNamespaces` so if any optional chain is not supported, it won't throw an error and just connect with the ones they support.

Trust already support this feature, and MM supports it partially (you have to manually change the network in their app to sign with Goerli)

```
export const sessionParams = {
  namespaces: {
    eip155: {
      methods: [
        'eth_sendTransaction',
        'eth_sign',
        'personal_sign',
        'eth_signTypedData',
      ],
      chains: ['eip155:1'],
      events: ['chainChanged', 'accountsChanged'],
      rpcMap: {},
    },
  },
  optionalNamespaces: {
    eip155: {
      methods: [
        'eth_sendTransaction',
        'eth_sign',
        'personal_sign',
        'eth_signTypedData',
      ],
      chains: ['eip155:5'],
      events: ['chainChanged', 'accountsChanged'],
      rpcMap: {},
    },
  },
};
```

**Reply by Lestonz** [reply]
Thank you very much, this worked for Trust W. I also solved the problem in Rainbow W.

Please, I would appreciate if you inform me with an announcement when there is a Metamask connection.

Additionally, please support to connect Sepolia Testnet shortly.

I will probably contact you again when I get errors while connecting Smart Contract and sending Transaction :)
